### PR TITLE
FilterSidebar functionality

### PR DESCRIPTION
**Pull request by Manju-sree**
### FilterSidebar functionality
I have added the FilterSideBar component with sortbyprice,pricerange,sortbycategory,sortbyratings and included instock and fast-delivery filters functionality in Products page. It's still in the experimental phase. So ignore other files. Focus only on the below mentioned files. 

## Files to check
please check all jsx files.

## Files to Ignore:
please ignore all .css files, as they are reviewed from previous project repo.

Preview : https://deploy-preview-6--angry-dubinsky-3738ba.netlify.app/Products

### CODE REVIEW

**Review by JantuDeb — APPROVED** [review]
codes are clean and structured. I would suggest you to add smaller number of files in one PR

File: src/components/FilterSideBar/FilterSideBar.jsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            </li>
+            <div className="filter-divider"></div>
+            <li className="filter-subtitle">Category</li>
+            {categories.map((categoryName) => (
+              <li key={categoryName}>
+                <label className="form-label">
+                  <input
+                    type="checkbox"
+                    className="form-checkbox-field"
+                    checked={state.categoryName}
```
Comment (JantuDeb) [inline src/components/FilterSideBar/FilterSideBar.jsx]:
```
checked = {state.category[categoryName]}
```

File: src/components/FilterSideBar/FilterSideBar.jsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                    type="radio"
+                    className="form-checkbox-field"
+                    name="rating"
+                    id="Stars & above"
+                    onChange={() => {
+                      dispatch({
+                        type: "FILTER_RATING",
+                        payload: { starCount: num },
+                      });
+                    }}
```
Comment (JantuDeb) [inline src/components/FilterSideBar/FilterSideBar.jsx]:
```
checked = {state.rating ===num}
```

File: src/App.js [hunk 3]
```diff
@@ -1,19 +1,19 @@
 import {Home,Products,Cart,WishList} from "./pages/index";
```
Comment (sushil-mod) [inline src/App.js:1]: I like the way you used one file for all components

File: src/context/FilterContext/FilterProvider.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+const FilterProvider = ({ children }) => {
+    const [productList, setProductList] = useState([]);
+    const [state, dispatch] = useReducer(filterReducerFunc, initialValues);
+    
+  
+console.log(state);
+    useEffect(() => {
+        (async () => {
+            try {
+                const { data: { products } } = await axios.get("/api/products");
```
Comment (sushil-mod) [inline src/context/FilterContext/FilterProvider.jsx:17]: i like how you destructured  data from response

**Review by sushil-mod — APPROVED** [review]
good code ,  i learned 2 new things from your code ,some filter are not reset after clear rest is good

**Comment by Manju-sree** [comment]
> good code , i learned 2 new things from your code ,some filter are not reset after clear rest is good

it's resolved now.

File: src/components/FilterSideBar/FilterSideBar.jsx (hunk 1 above)
Comment (Manju-sree) [reply src/components/FilterSideBar/FilterSideBar.jsx]: thanks a lot for solution

File: src/components/FilterSideBar/FilterSideBar.jsx (hunk 2 above)
Comment (Manju-sree) [reply src/components/FilterSideBar/FilterSideBar.jsx]: thanks for suggestion

**Pull request merged**